src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/CellUpdate.cs

```diff
@@ -0,0 +1,192 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Data.Common;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.EditData.UpdateManagement
+{
+    /// <summary>
+    /// Representation of a cell that should have a value inserted or updated
+    /// </summary>
+    public sealed class CellUpdate
+    {
+        private const string NullString = @"NULL";
+        private const string TextNullString = @"'NULL'";
+        private static readonly Regex HexRegex = new Regex("0x[0-9A-F]+", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        /// <summary>
+        /// Constructs a new cell update based on the the string value provided and the column
+        /// for the cell.
+        /// </summary>
+        /// <param name="column">Column the cell will be under</param>
+        /// <param name="valueAsString">The string from the client to convert to an object</param>
+        public CellUpdate(DbColumn column, string valueAsString)
+        {
```

Comment:
Please convert this 150 line constructor into many smaller methods - each handler should be its own method. This should just do the basic routing so it's clear what we're doing.